app/src/main/java/com/example/learnahead_prototyp/UI/HomeFragment.kt

```diff
@@ -41,4 +53,38 @@ class HomeFragment : Fragment() {
 
         binding.profile.setOnClickListener { findNavController().navigate(R.id.action_homeFragment_to_profileFragment) }
     }
+
+    private fun observer() {
+        // Beobachtet die LiveData-Objekte des SummaryViewModels und aktualisiert die UI entsprechend
+
+
+        // Beobachtet die LiveData-Objekte des AuthViewModels und aktualisiert die UI entsprechend
+        authViewModel.currentUser.observe(viewLifecycleOwner) { state ->
+            binding.progressBar.visibility = when (state) {
+                is UiState.Loading -> View.VISIBLE
+                is UiState.Failure -> {
+                    toast(state.error)
+                    View.GONE
+                }
+                is UiState.Success -> {
+                    currentUser = state.data
+                    View.GONE
+                }
+            }
+        }
+
+        authViewModel.updateUserInfo.observe(viewLifecycleOwner) { state ->
+            binding.progressBar.visibility = when (state) {
+                is UiState.Loading -> View.VISIBLE
+                is UiState.Failure -> {
+                    toast(state.error)
+                    View.GONE
+                }
+                is UiState.Success -> {
+                    toast("Du hast für deinen Login gerade $state erhalten.")
```

Comment:
Kann das funktionieren das $state???
Weil das ist doch nen User Objekt und du willst doch die Punkte zurückgeben

Reply:
$state ist hier UiState<String>, also ja das geht. Aber danke für die Info. $state.data wäre korrekt
```suggestion
                    toast("Du hast für deinen Login gerade $state.data erhalten.")
```